pkg/blob/controllerserver.go

```diff
@@ -466,16 +442,11 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Errorf(codes.Internal, "failed to create container(%s) on account(%s) type(%s) rg(%s) location(%s) size(%d), error: %v", validContainerName, accountName, storageAccountType, resourceGroup, location, requestGiB, err)
 	}
 	if volContentSource != nil {
-		dstAzcopyAuthEnv := srcAzcopyAuthEnv
-		dstAccountSASToken := srcAccountSASToken
-		if srcAccountName != accountName {
-			if dstAccountSASToken, dstAzcopyAuthEnv, err = d.getAzcopyAuth(ctx, accountName, accountKey, storageEndpointSuffix, accountOptions, secrets, secretName, secretNamespace); err != nil {
```

Comment:
this might have been causing your failures.

`getAzcopyAuth()` has a few issues:
- only tries to use MSI auth when `len(secrets) == 0 && len(secretName) == 0 `. 
- tries to get storage access key and sas token as a fallback even if disabled on the account
- uses the cluster identity to get the fallback key / sas which requires even more permissions

All of this account key / sas handling seems like hacks around the root issue that [the default permissions for the control plane identity](https://learn.microsoft.com/en-us/azure/aks/use-managed-identity#summary-of-managed-identities-used-by-aks) neglected to include `Storage Blob Data Contributor` along with `Contributor`.

Changing `secrets, secretName, secretNamespace` to `nil, "", ""` would be a quick fix, and preferably `getAzcopyAuth` would always try the identity first.

Reply:
we could address this in other PR, this is refactor not bug fix

Reply:
I also tried again and `Reader` can be worked after it failed before. But I think the reason is not here since we have use MSI in that test since `len(secrets) == 0 && len(secretName) == 0 `, I will still work on this issue to find the reason.